### ISSUE TITLE
Autocategorize based on Steam store tags

### ISSUE THREAD

**Issue by rallion** [rallion/depressurizer#3]
Method would be to create a list of tags, sorted by popularity, with the more popular tags first. The user would then select the tags they wanted applied as categories from a list. The ones that they are more likely to want to use would be found close to the beginning of this list.

Tags that match store genre names could optionally be filtered out to avoid clutter at the top of the list.

**Comment by alidan** [comment]
I dont know if this is the right place, but last time I looked at depressurizer it couldn't handle multiple tags, now it can. 
take a look at borderlands 2 and batman arkham games, they come up in the same category as each other because steam has no FPS genre. 
What im saying is I believe the priority for this should probably be higher than low at this point. sort by genre is good, but sort by tags would probably be as good as it can possibly get. 

for a frame of reference, i have over 750 games and just genre barely storts them.

**Comment by rallion** [comment]
The reason it's marked as Low is because I couldn't think of a way to make it as useful as it sounds. Just using the first tag or few tags would generally get you farther from your ideal category setup, not closer to it. It would be nice to put Borderlands in the FPS category, but do you also want to put Arkham Asylum in the Batman category at the same time? Is that really useful from a library browsing perspective?

However, as I was typing up a long explanation as to why, I thought of a different method that _might_ work acceptably. So, medium it is.

**Comment by alidan** [comment]
in all honesty, after the initial launch of tags that had games like the putt putt get tagged with "hard core rape simulator" the tags got usefull, lets use borderlands 2 as an example. 

it has fps, co-op, rpg, action, shooter, multiplayer, action rpg, adventure, online co-op, singleplayer
as general tags
it also has 
loot - basically think diablo style games with that
funny, open world, comedy, memes, stylized, futuristic, sci-fi, dark humor, atmospheric
all of which are fantastic ways to describe the game, and all of which are generally tags that i would look up for a game i'm in the mood to play. 

looking up batman arkham asylum all the tags are fantastic ways to describe the game. sure some of the tags are going to be useless, but there is a way to just remove a category that you don't like right? if thats not a feature already in the program, that may also be something to look into.

**Comment by rallion** [comment]
You can certainly remove categories. You can only do one at a time at the moment, but I already have plans to fix that.

The problem that you would run into with applying all tags as categories is that you could literally end up with more categories than you have games, even if you have hundreds of games. That's just not useful, and would take an unreasonable amount of effort to sort out. There needs to be a way to limit it before it gets that far.

**Comment by alidan** [comment]
ok i can see that, and for smaller libraries that could be an issue, i just went through quite a few just looking at tags, bastion is one that i think has 3 unhelpful tags and one that is game specific, awesomenauts have a few too, and apart from that i saw at most 10 tags that i would think are kind of useless when describing a game, at least from an "im in the mood to play X kind of game" standpoint. 

if you want my idea for it... have tag sorting a check box so you can't have it auto tag, in the tag have 3 settings, 
(everything) (user specified) (abridged) 
everything would do what it says, dump every tag into the categories
user specified would have users blacklist tags themselves and come up with their own prefered tags possibly mixed with abridged to start out with
abridged would just have the tags you decided to be most useful in it.

you could also have a setting to z the tags. while sorting things, i generally have things labeled with (- ) or (z ) depending on where i want them to appear in sort order, it works in so many applications to sort things to the very top and the very bottom that its something i always do. in the case of tags, adding a z with a space will force them below genres so they don't get mixed up in the event that someone wants the genres and tags to be in separate places. adding a - space to the tag will also make it rise to the top. granted both these are superseded by favorites and games, but its still a further method to sorting. 

but like i said above, this is all coming from someone who has 750+ games, there is tons of overlap and looking at all the games today, i can see a very minimal amount of tags that are undesirable, its not for someone with a manageable amount of games, but if i remember right this program wasnt made for them either.

**Comment by pdc1** [comment]
I found this program looking for a way to categorize using tags, so +1!

I've got 300+ games, and the Steam categories don't come near to cutting it. On the Steam side, Octodad and Medal of Honor are both "Action", which for my purposes is useless, but in particular I have a LOT of indie games (thanks to bundle-mania) and I want to create categories like "Indie Comedy" (Octodad) or "Military Shooter" (Medal of Honor). Okay, to be honest it's hard to categorize Octodad :)

My thought was let me create a bunch of steam categories, and then let me assign which tags must be present to use this category. That will probably autocat 90% and I can manually fix the outliers.

At this point it would be helpful just to have a list of all my games with their associated tags so I can start looking for patterns to create categories. Back to google...
